sandbox/lua/modules/line_protocol.lua

```diff
@@ -0,0 +1,246 @@
+    that utilize this module.  This function takes no arguments.
+
+    *Arguments*
+        - config (table or nil)
+            Table of config option overrides that come from the client
+            of this API.  Defaults for this module are defined within
+            the set_config function and clients implementing this API
+            can reference it for available options.
+
+    *Return*
+        A table of InfluxDB line protocol messages ready to be sent to
+        an InfluxDB server after being looped through in an encoder
+        implementing this API.
+
+--]=]
+
+local interp = require "msg_interpolate"
+local field_util = require "field_util"
+local math = require "math"
+local read_config = read_config
+local read_message = read_message
+local read_next_field = read_next_field
+local pairs = pairs
+local string = require "string"
+local table = require "table"
+local tostring = tostring
+local type = type
+
+local M = {}
+setfenv(1, M) -- Remove external access to contain everything in the module.
+
+local function influxdb_kv_fmt(string)
+    return tostring(string):gsub("([ ,])", "\\%1")
+end
+
+local function points_tags_tables(config)
+    local name_prefix = ""
+    if config["name_prefix"] then
+        name_prefix = interp.interpolate_from_msg(config["name_prefix"])
+    end
+    local name_prefix_delimiter = config["name_prefix_delimiter"] or ""
+    local used_tag_fields = config["used_tag_fields"]
+    local skip_fields = config["skip_fields"]
+
+    -- Initialize the tags table, including base field tag values in
+    -- list if the magic **all** or **all_base** config values are
+    -- defined.
+    local tags = {}
+    if not config["carbon_format"]
+        and (config["tag_fields_all"]
+        or config["tag_fields_all_base"]
+        or config["used_tag_fields"]) then
+            for field in pairs(field_util.base_fields_tag_list) do
+                if config["tag_fields_all"] or config["tag_fields_all_base"] or used_tag_fields[field] then
+                    local value = read_message(field)
+                    table.insert(tags, influxdb_kv_fmt(field).."="..tostring(influxdb_kv_fmt(value)))
+                end
+            end
+    end
+
+    -- Initialize the table of data points and populate it with data
+    -- from the Heka message.  When skip_fields includes "**all_base**",
+    -- only dynamic fields are included as InfluxDB data points, while
+    -- the base fields serve as tags for them. If skip_fields does not
+    -- define any base fields, they are added to the fields of each data
+    -- point and each dynamic field value is set as the "value" field.
+    -- Setting skip_fields to "**all_base**" is recommended to avoid
+    -- redundant data being stored in each data point (the base fields
+    -- as fields and as tags).
+    local points = {}
+    while true do
+        -- Iterate through Fields array in the message
+        local typ, name, value = read_next_field()
```

Comment:
Hmm, I'm still getting a null result from `read_message("raw")`, resulting in this error message:
```
2015/07/04 00:15:45 Plugin 'LogOutput' error: Error encoding message: FATAL: process_message() 
.../local/heka/share/heka/lua_modules/line_protocol.lua:135:bad argument #0 to 'decode_message'
(must have one string argument)
```
I'm using the 0.9.2 release for testing these modules/encoders.  Has this functionality been updated since that release?  I can try with the dev branch and report back with my findings.